### PR TITLE
fix(oomagent/join): fix deadlock

### PR DESCRIPTION
**Pull request by lianxmfor**
resolve: #1092

### CODE REVIEW

**Review by jinghancc — APPROVED** [review]
LGTM

**Review by YikSanChan — APPROVED** [review]
lgtm. does it exist a test case that test this?

File: oomagent/server.go [hunk 1]
```diff
@@ -243,33 +247,37 @@ func (s *server) ChannelJoin(stream codegen.OomAgent_ChannelJoinServer) error {
...
+				globalErr = err
+				break
+			}
+			// Only need to send header upon the first response
+			header = nil
 		}
 	}
 
-	close(entityRows)
 	// wait until oomstore ChannelJoin is done, whether succeeded or failed
```
Comment (YikSanChan) [inline oomagent/server.go:279]: is the comment outdated?

Comment (lianxmfor) [reply oomagent/server.go]: updated.

**Pull request merged**